aztec/src/main/kotlin/org/wordpress/aztec/watchers/event/sequence/known/space/API26InWordSpaceInsertionEvent.java

```diff
@@ -0,0 +1,147 @@
+    public boolean isUserOperationPartiallyObservedInSequence(@NotNull EventSequence<TextWatcherEvent> sequence) {
+        for (int i = 0; i < sequence.size(); i++) {
+
+            TextWatcherEvent eventHolder = getSequence().get(i);
+            TextWatcherEvent observableEvent = sequence.get(i);
+
+            // if time distance between any of the events is longer than 50 millis, discard this as this pattern is
+            // likely not produced by the platform, but rather the user.
+            // WARNING! When debugging with breakpoints, you should disable this check as time can exceed the 50 MS limit and
+            // create undesired behavior.
+            if (i > 0) { // only try to compare when we have at least 2 events, so we can compare with the previous one
+                long timestampForPreviousevent = sequence.get(i-1).getTimestamp();
+                long timeDistance = observableEvent.getTimestamp() - timestampForPreviousevent;
+                if (timeDistance > MAXIMUM_TIME_BETWEEN_EVENTS_IN_PATTERN_MS) {
+                    return false;
+                }
+            }
+
+            eventHolder.setBeforeTextChangedEvent(observableEvent.getBeforeEventData());
+            eventHolder.setOnTextChangedEvent(observableEvent.getOnEventData());
+            eventHolder.setAfterTextChangedEvent(observableEvent.getAfterEventData());
+
+            // items in this.getSequence() should always be ITextWatcherEventComparator,
+            // but we check to avoid any potential problems
+            if (!(eventHolder instanceof ITextWatcherEventComparator
+                    && ((ITextWatcherEventComparator)eventHolder).testFitsBeforeOnAndAfter())) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @NotNull
+    @Override
+    public TextWatcherEvent buildReplacementEventWithSequenceData(EventSequence<TextWatcherEvent> sequence) {
+        TextWatcherEventInsertText replacementEvent = new TextWatcherEventInsertText();
+        // here make it all up as a unique event that does the insert as usual, as we'd get it on older APIs
+        TextWatcherEvent firstEvent = sequence.get(0);
+        TextWatcherEvent lastEvent = sequence.get(sequence.size()-1);
+
+        BeforeTextChangedEventData beforeData = firstEvent.getBeforeEventData();
+
+        int differenceIndex = StringUtils.indexOfDifference(beforeData.getTextBefore(), lastEvent.getAfterEventData().getTextAfter());
+        replacementEvent.setInsertionStart(differenceIndex);
+        replacementEvent.setInsertionLength(1);
+
+        SpannableStringBuilder oldText = beforeData.getTextBefore();
+        oldText.insert(differenceIndex, SPACE_STRING);
+
+        replacementEvent.setAfterTextChangedEvent(new AfterTextChangedEventData(oldText));
+
+        return replacementEvent;
+    }
+
+    private void init() {
+        // here we populate our model of reference (which is the sequence of events we expect to find)
```

Comment:
Can't this code be moved to the constructor of the class?

Reply:
Totally - addressed in c21ffaa.
Part of the problems I had while attempting Java/Kotlin conversions :)